Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h

```diff
@@ -448,6 +454,7 @@ class GPUConnectionToWebProcess
     // GPU preferences don't change for a given WebProcess. Pages that use different GPUProcessPreferences
     // cannot be in the same WebProcess.
     const GPUProcessPreferencesForWebProcess m_preferences;
```

Comment:
Is this still needed?

Reply:
I thought I had added `GPUProcessPreferencesForWebProcess` purely for the purpose of IPC filtering.

Reply:
There are still a few preferences there:

 - bool isDOMRenderingEnabled { false };
 - bool isDynamicContentScalingEnabled { false };
 - bool allowTestOnlyIPC { false };
 - bool isLockdownSafeFontParserEnabled { false };

Reply:
Seems annoying to have 2 structs for similar things. We essentially want to share a setting to the GPUProcess or the NetworkProcess. Maybe `filtersIPC` should be renamed to `shareWith: NetworkProcess|GPUProcess` and then replace `GPUProcessPreferencesForWebProcess ` (and the network equivalent) entirely. Or something like `crossProcess: true` if we feel we don't need to distinguish GPUProcess and NetworkProcess and just share these particular settings with both of them.

Reply:
We can eventually move to always using this new mechanism but I don't want to do that in this PR because it's already pretty big as is. And I think we want these settings to be shared across processes. Differentiating between processes will induce an unnecessary complexity.

Reply:
> Differentiating between processes will induce an unnecessary complexity.
I agree and one of my 2 proposals addressed this.

I'm personally not sure that landing the patch as is would be great. It adds complexity since it adds one more way of doing something, rather than replacing the existing thing. Also, the name of the property in the yaml file wouldn't make sense if we extended this to properties that are shared with other processes but are NOT used for IPC filtering.

Reply:
What name do you suggest then? I really don't wanna waste anymore time spending my time renaming things. It took me a whole day to do this rename.

Reply:
Since this is a replacement for `GPUProcessPreferencesForWebProcess` and `NetworkProcessProcessPreferencesForWebProcess`, maybe `SharedPreferencesForWebProcess`?